src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -133,40 +138,67 @@ public InventorySync(@Nonnull final InventorySyncOptions syncOptions) {
         List<InventoryEntryDraft> accumulator = new ArrayList<>(syncOptions.getBatchSize());
         final List<CompletableFuture<Void>> completableFutures = new ArrayList<>();
         for (InventoryEntryDraft draft : drafts) {
-            if (draft != null) {
-                if (isNotEmpty(draft.getSku())) {
-                    accumulator.add(draft);
-                    if (accumulator.size() == syncOptions.getBatchSize()) {
-                        final CompletableFuture<Void> batchProcessingFuture =
-                            processBatch(accumulator, supplyChannelKeyToId)
-                                .exceptionally(exception -> null)
-                                .toCompletableFuture();
-                        completableFutures.add(batchProcessingFuture);
-                        accumulator = new ArrayList<>(syncOptions.getBatchSize());
-                    }
-                } else {
-                    statistics.incrementProcessed();
-                    statistics.incrementFailed();
-                    syncOptions.applyErrorCallback(INVENTORY_DRAFT_HAS_NO_SKU, null);
-                }
-            } else {
-                statistics.incrementProcessed();
-                statistics.incrementFailed();
-                syncOptions.applyErrorCallback(INVENTORY_DRAFT_IS_NULL, null);
+            if (shouldProceedWithDraft(draft)) {
+                accumulator.add(draft);
+                accumulator = syncAccumulator(accumulator, completableFutures, supplyChannelKeyToId, true);
```

Comment:
Not sure, if this is what you've asked for? I believe it's less confusing 9d38a3ff4c88df26cf70156c77f96b4a32639009